### ISSUE TITLE
reconsider which parts of rethrown errors are most important to surface

### ISSUE THREAD

**Issue by simonpcouch** [tidymodels/tune#599]
The improved errors in the new recipes release call out which step an error occurred in:

```
Error in `step_bs()`:
Caused by error in `if (...) NULL`:
! missing value where TRUE/FALSE needed
```

Many improved errors in upcoming tidyr and dplyr use similar machinery. Those are effectively surfaced in parallel (see https://github.com/tidymodels/tune/pull/598), but are obscured by the cataloger:

```
→ 1 | error: 
Caused by error in `if (...) NULL`:
! missing value where TRUE/FALSE needed
There were issues with some computations   1: x2
````

I'd anticipate this is because the cataloger uses `conditionMessage` while the previous logging machinery collapses messages manually:

https://github.com/tidymodels/tune/blob/a0fe1027b1bf24813596ae8f378933586d021cab/R/logging.R#L302-L307

**Comment by simonpcouch** [comment]
A related thought here—with an eye for `dplyr::last_dplyr_warnings()` and `.Last.tune.result`, we could retain the full catalog in `.Last.tune.catalog` or similar so that folks know where errors occurred and get access to the full message.

**Comment by simonpcouch** [comment]
Closing in favor of #610 and tidymodels/recipes#1080.
